File: ruby/007/main.rb
=begin
    ProjectEuler 007
    Ruby / main.rb
    copyright (c) 2015 Susisu

    Ruby 2.0.0p481
    `ruby main.rb`
=end

def main
    counter = 0
    n = 2
    while true do
        if prime? n then
            counter += 1
            if counter == 10001 then
                puts n
                break
            end
        end
        n += 1
    end
end

def prime?(n)
    if n < 2 then
        return false
    else
        r = Math.sqrt(n)
        i = 2
        while i <= r do
            if n % i == 0 then
                return false
            end
            i += 1
        end
        return true
    end
end

main()

File: ocaml/007/main.ml
(*
    ProjectEuler 007
    OCaml / main.ml
    copyright (c) 2015 Susisu

    OCaml 4.02.1
    `ocamlc main.ml -o bin/main`
*)

let is_prime n =
    let rec loop n m =
        if float_of_int m > sqrt (float_of_int n) then
            true
        else if n mod m = 0 then
            false
        else
            loop n (m + 1)
    in if n < 2 then false else loop n 2;;

let rec calc n c =
    if is_prime n then
        if c = 10000 then n else calc (n + 1) (c + 1)
    else
        calc (n + 1) c;;

let main () = begin
    print_int (calc 1 0);
    print_newline ()
end;;

main ();;
